packages/android_alarm_manager/test/android_alarm_manager_test.dart

```diff
@@ -0,0 +1,203 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:android_alarm_manager/android_alarm_manager.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  setUpAll(() {
+    AndroidAlarmManager.channel
+        .setMockMethodCallHandler((MethodCall call) => null);
+  });
+
+  test('${AndroidAlarmManager.initialize}', () async {
+    AndroidAlarmManager.channel
+        .setMockMethodCallHandler((MethodCall call) async {
+      assert(call.method == 'AlarmService.start');
+      return true;
+    });
+
+    final bool initialized = await AndroidAlarmManager.initialize();
+
+    expect(initialized, isTrue);
+  });
+
+  group('${AndroidAlarmManager.oneShotAt}', () {
+    test('validates input', () async {
+      final DateTime validTime = DateTime.utc(1993);
+      final int validId = 1;
+      final Function validCallback = (int id) => null;
+
+      // Callback should take a single int param.
+      expect(
+          () => AndroidAlarmManager.oneShotAt(
+              validTime, validId, ((String foo) => foo)),
```

Comment:
I wouldn't pass a closure to this method, even if it's valid for this test. Callbacks can only be tear-offs for static methods (top level or class level) so this gives a false impression that anonymous closures are supported.